north/src/command_processor.rs

```diff
@@ -0,0 +1,386 @@
+        if let Some(context) = app.process_context() {
+            container_info.set_running_pid(context.process().pid());
+        }
+        container_info.set_version(app.version().to_string());
+        if app.container().is_resource_container() {
+            container_info.set_field_type(api::ContainerInfo_Type::RESOURCE);
+        } else {
+            container_info.set_field_type(api::ContainerInfo_Type::APP);
+        }
+        list_response.containers.push(container_info);
+    }
+    response.set_container_list(list_response);
+    Ok(response)
+}
+
+/// List all running applications.
+async fn ps(state: &State) -> Result<api::Response> {
+    let mut response = api::Response::new();
+    let mut list_response = api::ProcessList::default();
+    for app in state.applications().sorted_by_key(|app| app.name()) {
+        let mut process_info = api::ProcessInfo::default();
+        if let Some(ref context) = app.process_context() {
+            process_info.set_name(app.name().to_string());
+            process_info.set_pid(context.process().pid());
+            process_info.set_version(app.version().to_string());
+            process_info.set_uptime(context.uptime().as_nanos() as i64);
+
+            #[cfg(any(target_os = "android", target_os = "linux"))]
+            {
+                const PAGE_SIZE: usize = 4096;
+                let pid = context.process().pid();
+                let statm = procinfo::pid::statm(pid as i32)?;
+                process_info.set_size((statm.size * PAGE_SIZE) as u64);
+                process_info.set_resident((statm.resident * PAGE_SIZE) as u64);
+                process_info.set_shared((statm.share * PAGE_SIZE) as u64);
+                process_info.set_text((statm.text * PAGE_SIZE) as u64);
+                process_info.set_data((statm.data * PAGE_SIZE) as u64);
+            }
+            list_response.mut_processes().push(process_info);
+        }
+    }
+    response.set_process_list(list_response);
+    Ok(response)
+}
+
+/// Start applications. If no regex string is provided, *all* known applications that
+/// are not in a running state are started. If a regex string is supplied it
+/// is used to construct a Regex and all container (names) matching that
+/// Regex are attempted to be started.
+async fn start(state: &mut State, regex_str: Option<&str>) -> Result<api::Response> {
+    let mut response = api::Response::new();
+    let mut started_containers = api::StartedResponse::default();
+    let startup_list = started_containers.mut_startup_list();
+    let re = id_to_regex(regex_str)?;
+
+    let apps = state
+        .applications()
+        // Filter for not already running containers
+        .filter(|app| app.process_context().is_none())
+        // Filter ressource container that are not startable
+        .filter(|app| !app.container().is_resource_container())
+        // Filter matching container
+        .filter(|app| re.is_match(app.name()))
```

Comment:
Maybe we should check for a **full** match.
So that if you have a `hello` and `ferris_says_hello` container a `start hello` command does not surprisingly start both...
(If required, then you would need to run `start .+hello.+` then)